### PR TITLE
[mixer-e2e-test] add retry to prometheus query in check cache test

### PR DESCRIPTION
**Pull request by bianpengyuan**
Make prometheus sync period configurable so that check cache could sleep a little bit longer.

Another attempt to fix check cache test flake: #10786 

/assign @douglas-reid 
/assign @utka

### CODE REVIEW

**Comment by utka** [comment]
Thought you increased wait time, the test still failed with prometheus erroring :(

**Comment by bianpengyuan** [comment]
@utka Do you mean pod not starting up error? I don't think this change will help with that. This is to deflake a specific test in mixer e2e, which is caused by prometheus metric not updated in time.

**Comment by utka** [comment]
Oops I was thinking about another bug

File: tests/e2e/tests/mixer/mixer_test.go [hunk 1]
```diff
@@ -869,19 +867,29 @@ func testCheckCache(t *testing.T, visit func() error, app string) {
...
+		t.Helper()
+		t.Logf("Trying to query new check cache hit (attempt %d)...", i)
+		got, err := getCheckCacheHits(promAPI, app)
+		if err != nil {
+			errorf(t, "unable to retrieve valid cached hit number: %v", err)
+			return err
+		}
+		t.Logf("New cache hits: %v", got)
+		if got == prior {
+			return errors.New("got check cache hit 0 want at least 1")
```
Comment (douglas-reid) [inline tests/e2e/tests/mixer/mixer_test.go]: should this error message use `prior` and `got`, instead of hardcoding `0` and `1`?

**Comment by bianpengyuan** [comment]
gentle ping.

**Pull request merged**